pkg/cmd/server/origin/master_config.go

```diff
@@ -168,22 +193,40 @@ func BuildMasterConfig(options configapi.MasterConfig) (*MasterConfig, error) {
 		PrivilegedLoopbackClientConfig:     *privilegedLoopbackClientConfig,
 		PrivilegedLoopbackOpenShiftClient:  privilegedLoopbackOpenShiftClient,
 		PrivilegedLoopbackKubernetesClient: privilegedLoopbackKubeClient,
+
+		ServiceAccountPrivateKeyFile: options.ServiceAccountConfig.PrivateKeyFile,
+		ServiceAccountManagedNames:   options.ServiceAccountConfig.ManagedNames,
 	}
 
 	return config, nil
 }
 
-func newAuthenticator(servingInfo configapi.ServingInfo, etcdHelper tools.EtcdHelper, apiClientCAs *x509.CertPool) authenticator.Request {
-	tokenAuthenticator := getEtcdTokenAuthenticator(etcdHelper)
-
+func newAuthenticator(config configapi.MasterConfig, etcdHelper tools.EtcdHelper, tokenGetter serviceaccount.ServiceAccountTokenGetter, apiClientCAs *x509.CertPool) authenticator.Request {
```

Comment:
Seems like you should tighten this down to ServiceAccountConfig and OAuthConfig.